Syntax Errors + updated Ticket

xcoga opened the pull request:
Syntax errors in comments
updated Ticket class isAHoliday() method to compare showtime with holiday by converting LocalDateTime to LocalDate by java method.

xcoga committed:
syntax in ci cinemalcontroller

need 2 parameters for cinemacontroller ci

xcoga committed:
no static in interface

1) not allowed to use static in interface, ide suggested using public boolean login()
did not modify the method, waiting for further comments
2) Syntax error in createAccount(), updated Arraylist<Staff> to ArrayList<Staff> ( 'L' was not capitalised)

xcoga committed:
Syntax errors

1) IDE suggested adding @SuppressWarnings("serial") for this class
2) Line 47
//auth is non static, cannot be used in a static method. IDE suggests declaring auth as static or make method 'non-static'
3) Line 191
The ticket created here does not match the parameters of ticket constructor
4) Line 261
//the method params here does not match the params declared in cinema 'public void printLayout(String title, LocalDateTime showtime)'
5) line 284
//is this 'i' supposed to be 'index'? variable 'i' was not created

xcoga committed:
syntax

1) Ide suggested @SuppressWarnings("serial")
2)line 18, was this a typo? object was not used.

xcoga committed:
syntax

1) line 33
//same as customer class, auth has to be static in a static method. either that or method becomes non static
2) line 51
//ide complains that tempUsername should be an object, not string, suggests .contains() method the parameter should be <Staff> object.
3) Line 126 
mvc object was not created
4) need to import
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;
5) line 168, line 301, line 481 
IDE suggested to close the scanner item, because of resource leak

xcoga committed:
changed localdate to localdatetime

xcoga committed:
Updated isAHoliday() method

updateed method so as to get LocalDate type from LocalDateTime to compare with Holiday that is LocalDate type

xcoga committed:
change to LocalDate type

reverted my own changes previously, where i changed to LocalDateTime

aryans1204 committed:
Update TestingClient.java

aryans1204 committed:
Update Cinema.java

aryans1204 committed:
Update Customer.java

aryans1204 committed:
Update Customer.java

aryans1204 commented on src/entities/Customer.java:
```diff
@@ -185,6 +188,8 @@ private String makeBooking(ArrayList<Cinema> cinemaDB, ArrayList<Movie> movieDB)
 			String TID = Integer.toString(cinema.getName().hashCode()%1000) + showtime.toString();
+			
+			//Constructor has been updated, creation of ticket here need to match constructor
 			Ticket newTicket = new Ticket(mov, cinema, this, showtime, cinema.getSeat(title+showtime.toString(), index), TID, this.username, this.email, this.mobileNumber);
```
I've made the edits based on your comments for other classes, but since you wrote Ticket, could you make the edit here and push another commit reflecting that?

aryans1204 committed:
Update Customer.java

aryans1204 committed:
Update Customer.java

aryans1204 committed:
Update Staff.java

aryans1204 committed:
Update Staff.java

aryans1204 committed:
Update Staff.java

aryans1204 reviewed (changes requested):
Just that one addition, then looks ready to merge. Re-request review from me once that is done, and I've already made all the other edits that were specified in the comments.

xcoga committed:
updated holidaycontroller

only left the search, update, getall and getfile, setfilename in controller

xcoga committed:
added updateholiday in staffui

staffui will add/remove holidays

xcoga committed:
added method, update constructor

1) updated constructor to take in int of custmobileNumber, removed holiday arraylist from constructor
2) Changed how we will get holidays from DB
- TestingClient will set the Class Variable Holidays in Ticket after staffUI updates the HolidayDB
-  Class Variable holidays will then be used to compare with showtime

xcoga committed:
left out ' ;'

xcoga committed:
added try catch for case 10 in staffui

aryans1204 reviewed (approved):
LGTM 🚀
